app/controllers/chat_channels_controller.rb

```diff
@@ -181,4 +182,12 @@ def render_chat_channel
       render json: { errors: @chat_channel.errors.full_messages }
     end
   end
+
+  def send_open_notification
+    if @chat_channel.group?
+      Pusher.trigger("private-message-notifications-#{session_current_user_id}", "message-opened", { channel_type: @chat_channel.channel_type, adjusted_slug: @chat_channel.adjusted_slug }.to_json)
+    else
+      Pusher.trigger("private-message-notifications-#{session_current_user_id}", "message-opened", { channel_type: @chat_channel.channel_type, adjusted_slug: @chat_channel.adjusted_slug(current_user) }.to_json)
+    end
```

Comment:
this can be cleared up a little bit, the two pusher trigger lines are really long.

My suggestion:

```suggestion
    adjusted_slug = if @chat_channel.group?
                      @chat_channel.adjusted_slug
                    else
                      @chat_channel.adjusted_slug(current_user)
                    end

    Pusher.trigger(
      "private-message-notifications-#{session_current_user_id}",
      "message-opened",
      { channel_type: @chat_channel.channel_type, adjusted_slug: adjusted_slug }.to_json,
    )
```

Reply:
BTW reading this method seems that it's going to work only for direct channels, not for the other ones.

We have three types of channels: 

- `open` which are groups as General
- `invite only` 
- `direct`

as only `direct` channels are named `private-message-notifications-{USER_ID}`, what would happen to the other channels with this method?

If you look at https://github.com/thepracticaldev/dev.to/blob/0fb38adf48277b9b3f2b86245ed3e8d21a596bb8/app/controllers/messages_controller.rb#L121 you'll notice how messages are routed on different channels by calling `chat_channel.pusher_channels` in some case.

That might also be why I get this error when I test it with an open channel locally:

```js
 TypeError: "newMessages[activeChannelId][foundIndex] is undefined"
    value chat.jsx:736
    setState Preact
    value chat.jsx:731
chat.jsx:866:12
    value chat.jsx:866
```

I get `foundIndex` as `-1` which means that the message sent from Ruby wasn't found by this JS callback's part:

```javascript
          const foundIndex = messages[activeChannelId].findIndex(
            message => message.temp_id === response.message.temp_id,
          );
```

If all channels [need to require Pusher authentication](https://pusher.com/docs/channels/using_channels/private-channels) we need to renamed then `private-` something. Should they be renamed in the Pusher console as well as the DB? Because right now, my open channel for example, they have names like:

```ruby
[14] pry(main)> cc.pusher_channels
=> "open-channel-5"
```

You can create your own open channel with `ChatChannel.create_with_users(User.all, "open")` to test this hypothesis

Reply:
I found the real issue and it's because of something else. Something in the frontend. Will send an update soon.

Reply:
Also to explain how the sockets are handled using pusher in connect, take a look at this flow chart. This shows a message with mention works.

<img width="902" alt="Screenshot 2020-01-23 at 2 47 06 PM" src="https://user-images.githubusercontent.com/3650216/72971336-7fdeb000-3def-11ea-9d63-3c68620d210a.png">

though I have ideas on how to make this more efficient in the future.

Reply:
Hi @sarthology, thanks for the info! It seems to work a little bit better, though now I have to refresh the page to see the sent message in the chat. See the recording below.

Your reply did not address the other concerns though, have you tested the PR with all types of channels? Because the code seems geared only towards private channels. See my notes above.

![chat](https://user-images.githubusercontent.com/146201/72972484-d1754380-3dcb-11ea-913f-a31e1cdcba68.gif)

As you can see in the recording, when I type a message it sends `POST /messages` but the message is not added to the inline chat, it only appears when I refresh the whole page.

Reply:
Yes I have tested it with various channels. Also the new error is also not repopulating.

![Jan-23-2020 15-15-51](https://user-images.githubusercontent.com/3650216/72973449-490a9900-3df3-11ea-9f09-b7a9cc92d5da.gif)